src/query-string-arr.js

```diff
@@ -0,0 +1,163 @@
+const strictUriEncode = require('strict-uri-encode')
+const decodeComponent = require('decode-uri-component')
+
+// code heavily borrowed from https://github.com/sindresorhus/query-string
+
+function parserForArrayFormat() {
```

Comment:
we should use arrow functions for these definitions because none of these functions reference themselves. I also want to stay consistent with the rest of the package